active/User defined attacks.js

```diff
@@ -0,0 +1,202 @@
+	"SQLException",
+	"supplied argument is not a valid",
+	"Syntax error in query expression",
+	"The error occurred in",
+	"The script whose uid is",
+	"Type mismatch",
+	"Unable to jump to row",
+	"Unclosed quotation mark before the character string",
+	"unexpected end of SQL command",
+	"unexpected error",
+	"Unterminated string constant",
+	"Warning: mysql_query",
+	"Warning: pg_connect",
+	//"Warning: pg_connect\(\): Unable to connect to PostgreSQL server: FATAL",
+	"You have an error in your SQL syntax near",
+]
+
+/**
+ * Scans a "node", i.e. an individual entry in the Sites Tree.
+ * The scanNode function will typically be called once for every page. 
+ * 
+ * @param as - the ActiveScan parent object that will do all the core interface tasks 
+ *     (i.e.: sending and receiving messages, providing access to Strength and Threshold settings,
+ *     raising alerts, etc.). This is an ScriptsActiveScanner object.
+ * @param msg - the HTTP Message being scanned. This is an HttpMessage object.
+ */
+function scanNode(as, msg) {
+	// Do nothing here - this script just attacks parameters rather than nodes
+}
+
+/**
+ * Scans a specific parameter in an HTTP message.
+ * The scan function will typically be called for every parameter in every URL and Form for every page.
+ * 
+ * @param as - the ActiveScan parent object that will do all the core interface tasks 
+ *     (i.e.: sending and receiving messages, providing access to Strength and Threshold settings,
+ *     raising alerts, etc.). This is an ScriptsActiveScanner object.
+ * @param msg - the HTTP Message being scanned. This is an HttpMessage object.
+ * @param {string} param - the name of the parameter being manipulated for this test/scan.
+ * @param {string} value - the original parameter value.
+ */
+function scan(as, msg, param, value) {
+	// Debugging can be done using println like this
+	//println('scan called for url=' + msg.getRequestHeader().getURI().toString() + 
+	//	' param=' + param + ' value=' + value);
+	
+	var max_attacks = attacks.length	// No limit for the "INSANE" level ;)
+	
+	if (as.getAlertThreshold() == "LOW") {
+		max_attacks = 6
+	} else if (as.getAttackStrength() == "MEDIUM") {
+		max_attacks = 12
+	} else if (as.getAttackStrength() == "HIGH") {
+		max_attacks = 24
+	}
+
+	for (i in attacks) {
+		// Dont exceed recommended number of attacks for strength
+		// feel free to disable this locally ;)
+		if (i > max_attacks) {
+			return
+		}
+		// Copy requests before reusing them
+		msg = msg.cloneRequest();
+
+		// setParam (message, parameterName, newValue)
+		//println('Attacking with ' + attacks[i])
+		as.setParam(msg, param, attacks[i]);
+		
+		// sendAndReceive(msg, followRedirect, handleAntiCSRFtoken)
+		as.sendAndReceive(msg, false, false);
+
+		// Add any generic checks here, eg
+		var code = msg.getResponseHeader().getStatusCode()
+		if (code >= 500 && code < 600) {
+			raiseAlert(as, msg, param, attacks[i], code)
+			// Only raise one alert per node
```

Comment:
Should this comment say "param"? (Following with the comment in the scanNode function above.)